lib/core/test/unit/Cardano/Wallet/DB/Fixtures.hs

```diff
@@ -0,0 +1,181 @@
+import Data.Either
+    ( fromRight )
+import Data.Time.Clock
+    ( UTCTime )
+import Data.Time.Clock.POSIX
+    ( posixSecondsToUTCTime )
+import Database.Persist.Sql
+    ( deleteWhere, insert_ )
+import Database.Persist.Sqlite
+    ( SqlPersistT, (==.) )
+import Test.QuickCheck
+    ( Arbitrary
+    , Gen
+    , Property
+    , Testable
+    , arbitrary
+    , counterexample
+    , cover
+    , elements
+    , frequency
+    , scale
+    , suchThat
+    )
+import Test.QuickCheck.Monadic
+    ( PropertyM, assert, monadicIO, monitor, run )
+import UnliftIO.Exception
+    ( bracket )
+
+import qualified Cardano.Wallet.DB.Sqlite.Schema as TH
+
+{-------------------------------------------------------------------------------
+    DB setup
+-------------------------------------------------------------------------------}
+withDBInMemory :: (SqliteContext -> IO a) -> IO a
+withDBInMemory action = bracket newDBInMemory fst (action . snd)
+
+newDBInMemory :: IO (IO (), SqliteContext)
+newDBInMemory = newInMemorySqliteContext nullTracer [] migrateAll
+
+initializeWallet :: WalletId -> SqlPersistT IO ()
+initializeWallet wid = do
+    deleteWhere [TH.WalId ==. wid] -- triggers delete cascade
+    insertWalletTable wid
+
+-- | Insert a wallet table in order to satisfy  FOREIGN PRIMARY constraints
+insertWalletTable :: WalletId -> SqlPersistT IO ()
+insertWalletTable wid = insert_ $ Wallet
+    { walId = wid
+    , walName = "Stores"
+    , walCreationTime = dummyUTCTime
+    , walPassphraseLastUpdatedAt = Nothing
+    , walPassphraseScheme = Nothing
+    , walGenesisHash = BlockId dummyHash
+    , walGenesisStart = dummyUTCTime
+    }
+
+{-------------------------------------------------------------------------------
+    Arbitrary
+-------------------------------------------------------------------------------}
+dummyUTCTime :: UTCTime
+dummyUTCTime = posixSecondsToUTCTime 1506203091
+
+dummyHash :: Hash "BlockHeader"
+dummyHash = Hash $ unsafeFromHex
+    "5f20df933584822601f9e3f8c024eb5eb252fe8cefb24d1317dc3d432e940ebb"
+
+{-------------------------------------------------------------------------------
+    QuickCheck utilities
+-------------------------------------------------------------------------------}
+-- | Like 'assert', but allow giving a label / title before running a assertion
+assertWith :: String -> Bool -> PropertyM IO ()
+assertWith lbl condition = do
+    let flag = if condition then "✓" else "✗"
+    monitor (counterexample $ lbl <> " " <> flag)
+    assert condition
+
+-- | rescale logaritmically, any base
+logScale' :: Double -> Gen a -> Gen a
+logScale' b = scale (floor @Double . logBase b . fromIntegral . succ)
+
+-- | rescale logaritmically
+logScale :: Gen a -> Gen a
+logScale = logScale' $ exp 1
+
+-- | cover with use of ($)
+coverM :: (Functor f, Testable prop) => Double -> Bool -> String -> f prop -> f Property
+coverM n c t =fmap $ cover n c t
+
+-- | like frequency but use only one generator with different filters
```

Comment:
```suggestion
-- | Like 'frequency' but use only one generator with different filters.
```

Reply:
(By the way, note that `suchThat` discards values until the predicate is satisfied. This can slow down unit tests if overused.)

Reply:
Yep, thanks for reminding